.github/workflows/reusable-publish-docs.yml

```diff
@@ -0,0 +1,118 @@
+name: Reusable Publish docs
+
+env:
+  BRANCH: main
+  ORIGIN: aws-powertools/powertools-lambda-java
+  VERSION: ""
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        description: "Version to build and publish docs (1.28.0, develop)"
+        required: true
+        type: string
+      alias:
+        description: "Alias to associate version (latest, stage)"
+        required: true
+        type: string
+      detached_mode:
+        description: "Whether it's running in git detached mode to ensure git is sync'd"
+        required: false
+        default: false
+        type: boolean
+
+permissions:
+  contents: write
+  id-token: write
+  pages: write
+
+jobs:
+  publish-docs:
+    runs-on: ubuntu-latest
+    environment: Docs
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab  # v3.5.2
+        with:
+          # While `fetch-depth` is used to allow the workflow to later commit & push the changes.
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@57ded4d7d5e986d7296eab16560982c6dd7c923b # v4.6.0
+        with:
+          python-version: "3.8"
+      - name: Install doc generation dependencies
+        run: |
+          pip install --upgrade pip 
+          pip install -r docs/requirements.txt
+      - name: Setup doc deploy
+        run: |
+          git config --global user.name Docs deploy
```

Comment:
If we're not committing and pushing, we don't need to set the user.email or user.name

Reply:
There is a separate branch `gh-pages` for storing versioned docs as backup and we commit and push to it.